sdks/python/apache_beam/examples/inference/multi_language_inference/expansion_service/run_inference_expansion.py

```diff
@@ -0,0 +1,135 @@
+        """
+    def batch_elements_kwargs(self):
+      return {'max_batch_size': 1}
+
+  class Preprocess(beam.DoFn):
+    def __init__(self, tokenizer):
+      self._tokenizer = tokenizer
+      logging.info('Starting Preprocess.')
+
+    def process(self, text: str):
+      import torch
+      # remove unusable tokens marks.
+      removable_tokens = ['"', '*', '<br />', "'", "(", ")"]
+      for token in removable_tokens:
+        text = text.replace(token, '')
+
+      # only take first sentence.
+      ending_chars = ['.', '!', '?']
+      for char in ending_chars:
+        if char in text:
+          text = text.split(char)[0]
+
+      # add dot to end of sentence.
+      text = text + ' .'
+
+      # mask the last word and drop very long sentences.
+      if len(text.strip()) > 0 and len(text.strip()) < 512:
+        logging.info('Preprocessing Line: %s', text)
+        text_list = text.split()
+        masked_text = ' '.join(text_list[:-2] + ['[MASK]', text_list[-1]])
+        tokens = self._tokenizer(masked_text, return_tensors='pt')
+        tokens = {key: torch.squeeze(val) for key, val in tokens.items()}
+
+        # skip first row of csv file.
+        if "review,sentiment" not in text.strip():
+          return [(text, tokens)]
+
+  class Postprocess(beam.DoFn):
+    def __init__(self, bert_tokenizer):
+      self.bert_tokenizer = bert_tokenizer
+      logging.info('Starting Postprocess')
+
+    def process(self, element: typing.Tuple[str, PredictionResult]) \
+        -> typing.Iterable[str]:
+      text, prediction_result = element
+      inputs = prediction_result.example
+      logits = prediction_result.inference['logits']
+      mask_token_index = (
+          inputs['input_ids'] == self.bert_tokenizer.mask_token_id).nonzero(
+              as_tuple=True)[0]
+      predicted_token_id = logits[mask_token_index].argmax(axis=-1)
+      decoded_word = self.bert_tokenizer.decode(predicted_token_id)
+      text = text.replace('.', '').strip()
+      yield (
+          f"{text} \n Predicted word: {decoded_word.upper()} -- "
+          f"Actual word: {text.split()[-1].upper()}")
+
+  def __init__(self, model, model_path):
+    self._model = model
+    logging.info(f"Downloading {self._model} model from GCS.")
+    self._model_config = BertConfig.from_pretrained(self._model)
+    self._tokenizer = BertTokenizer.from_pretrained(self._model)
+    self._model_handler = self.PytorchModelHandlerKeyedTensorWrapper(
+        state_dict_path=(model_path),
+        model_class=BertForMaskedLM,
+        model_params={'config': self._model_config},
+        device='cuda:0')
+
+  def expand(self, pcoll):
+    return (
+        pcoll
+        | 'Preprocess' >> beam.ParDo(self.Preprocess(self._tokenizer))
+        | 'Inference' >> RunInference(KeyedModelHandler(self._model_handler))
+        | 'Postprocess' >> beam.ParDo(self.Postprocess(
+            self._tokenizer)).with_input_types(typing.Iterable[str]))
```

Comment:
@chamikaramj when do we need with_input_types? I would've expected iterable of str to get auto-inferred, but I'm not really sure.

Do we have clear docs on this? It would be nice to link to those from here